src/components/RichTextPopup/RichTextPopup.js

```diff
@@ -107,34 +117,49 @@ const RichTextPopup = () => {
     return format;
   };
 
-  const handleClick = format => () => {
+  const handleTextFormatChange = format => () => {
     const { index, length } = editorRef.current.getSelection();
     const currentFormat = editorRef.current.getFormat(index, length);
 
     applyFormat(format, !currentFormat[format]);
   };
 
+  const handleSymbolsClick = () => {
+    setSymbolsVisible(!symbolsVisible);
+  };
+
   const handleColorChange = (_, color) => {
-    applyFormat('color', color.toHexString());
+    applyFormat("color", color.toHexString());
   };
 
   const applyFormat = (formatKey, value) => {
     editorRef.current.format(formatKey, value);
 
-    if (formatKey === 'color') {
+    if (formatKey === "color") {
       value = new window.Annotations.Color(value);
     }
 
     // format the entire editor doesn't trigger the editorTextChanged event, so we set the format state here
     setFormat({
       ...format,
-      [formatKey]: value,
+      [formatKey]: value
     });
   };
 
   const syncDraggablePosition = (e, { x, y }) => {
     setDraggablePosition({ x, y });
   };
+
+  const insertSymbols = symbol => {
+    const contents = editorRef.current.getContents();
+    const { index, length } = editorRef.current.getSelection();
+    // only update the selected word (defined by index)
+    editorRef.current.setText(

```

Comment:
It looks like `setText` will remove all the formats. See the following gif:

![2020-08-07 10-30-51 2020-08-07 10_31_32](https://user-images.githubusercontent.com/25110092/89673036-879d0800-d89a-11ea-848f-a7f15ebd0246.gif)

On a user's perspective, I think the format of the symbol should be the format at the current selection index, and format of the rest of the text should be kept.

I haven't tried https://quilljs.com/docs/api/#inserttext but this API sounds like what we need?

Reply:
This is fixed, use `deleteText` and `insertText` to replace `setText` function, please take a look and let me know.